src/client/contexts/CubeContext.tsx

```diff
@@ -813,10 +829,7 @@ export function CubeContextProvider({
       const newChanges = deepCopy(changes);
 
       for (const edit of list) {
-        const editIndex = newChanges[edit.board]?.edits?.findIndex((e) => e.index === edit.index) || -1;
```

Comment:
The reason why reverting bulk removes left one item is because findIndex returned zero but since that is falsey the `|| -1` occurs and thus it doesn't splice the array